frontend/javascripts/dashboard/dataset/import_delete_component.js

```diff
@@ -55,6 +55,7 @@ const ImportDeleteComponent = ({ datasetId, history }: Props) => {
     setIsDeleting(false);
     // Trigger dataset check to make sure the dataset list is up-to-date
     // but also make sure that the toast can be read
+    // TODO it is no longer necessary to recheck the datasets since the delete call now removes the db from its cache
```

Comment:
Up until now, the datastore did not remove a deleted dataset from its cache directly after deleting, so it was necessary to check the disk. This is not necessary anymore and could be replaced by the less expensive `fetchDatasets` call. However, since we sleep for 2 seconds anyway, this would not improve the performance here. What are your opinions on this?

Reply:
I'm not sure why the sleep exists. Does the `checkDatasets` call block until the backend is done with checking the disk? If yes, the sleep is unnecessary. If not, the sleep might be insufficient as the checking can easily take more than 2 seconds? As you suggest, I'd replace `checkDatasets` with `fetchDatasets` and then remove the sleep altogether.